### ISSUE TITLE
Specifying xdebug2 and PHP 8 installs xdebug3 instead

### ISSUE THREAD

**Issue by danepowell** [shivammathur/setup-php#519]
**Describe the bug**
If you specify the following...
```yaml
- uses: shivammathur/setup-php@v2
   with:
     php-version: "8.0"
     coverage: xdebug2
```
You actually get PHP 8.0 and Xdebug 3. Since Xdebug 2 and PHP 8.0 are an invalid combination, I would expect an error to be generated instead of silently installing Xdebug 3.

**Version**
- [x] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**
Linux

**Comment by shivammathur** [comment]
Fixed in 2.16.0

Now, It will fail to set up Xdebug when `xdebug2` is specified as the coverage driver on PHP 8.0 and above. Same for `xdebug3` for PHP 7.1 and lower. It will leave an error in the logs.

```yml
- name: Setup PHP
  uses: shivammathur/setup-php@v2
  with:
    php-version: ${{ matrix.php }}
    coverage: xdebug2
```

If you want your workflow to fail at this error, you can set `fail-fast` environment variable to `true`.

```yml
- name: Setup PHP
  uses: shivammathur/setup-php@v2        
  env:
    fail-fast: true
  with:
    php-version: ${{ matrix.php }}
    coverage: xdebug2
```

**Comment by danepowell** [comment]
Thanks!
